SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialReceiver.cpp

```diff
@@ -2260,7 +2260,12 @@ void USpatialReceiver::ResolveObjectReferences(FRepLayout& RepLayout, UObject* R
 
 		if (ObjectReferences.Array)
 		{
-			check(Property->IsA<UArrayProperty>());
+			UArrayProperty* ArrayProperty = Cast<UArrayProperty>(Property);
+			if (ArrayProperty == nullptr)
```

Comment:
Why are we downgrading this from a `check`? The only way this can be hit is if the logic that populates object references is changed and this isn't updated so it would be a GDK developer error.

Reply:
I agree with your statement, can put the nullptr check there instead.

Reply:
The check is back.